tests/unit/test_task_result.py

```
@@ -0,0 +1,83 @@
+    ]
+    
+    with patch('kodeagent.kutils.call_llm', side_effect=assistant_sequence):
+        async for _ in agent.run('test task'):
+            pass
+            
+    assert agent.task.result == "42"
+    assert agent.task.is_finished is True
+
+
+@pytest.mark.asyncio
+async def test_task_result_stored_on_failure(mock_llm):
```

Ruff (0.14.10)

Unused function argument: `mock_llm`

(ARG001)